src/os/pl-prologflag.c

```diff
@@ -1323,6 +1323,7 @@ initPrologFlags(void)
   setPrologFlag("c_libplso", FT_ATOM, C_LIBPLSO);
   setPrologFlag("c_ldflags", FT_ATOM, C_LDFLAGS);
   setPrologFlag("c_cflags",  FT_ATOM, C_CFLAGS);
+  setPrologFlag("default_tmp_dir", FT_ATOM|FF_READONLY, SWIPL_DEFAULT_TMP_DIR);
```

Comment:
I'd be more in favor of `tmp_dir` and initialize it the same way as the system decides on its tmp dir.   Next, make the places where this is used use the Prolog flag.  Can you manage that (bit like `posix_shell`)?

Reply:
Sure, sounds great!

Reply:
Patch coming a little later.